packages/nimble-components/src/wafer-map/modules/experimental/prerendering.ts

```diff
@@ -0,0 +1,85 @@
+import { scaleLinear } from 'd3-scale';
+import { ticks } from 'd3-array';
+import { WaferMapColorScaleMode } from '../../types';
+import type { Dimensions, WaferRequiredTypeMap } from '../../types';
+import type { WaferMap } from '../..';
+
+/**
+ * Prerendering prepares render-ready dies data to be used by the rendering module
+ */
+export class Prerendering<T extends WaferRequiredTypeMap> {
+    public get labelsFontSize(): number {
+        return this._labelsFontSize;
+    }
+
+    public get colorScale(): {
+        colors: string[],
+        values: number[]
+    } {
+        return this._colorScale;
+    }
+
+    private _colorScale!: {
+        colors: string[],
+        values: number[]
+    };
```

Comment:
Why a type like this instead of doing something like:
```
interface IColorScaleMarker {
    color: string,
    value: number
}

export type ColorScale = IColorScaleMarker[];
```

Having two independent arrays that I would assume are supposed to be the same length seems an unnecessary type decision.

Reply:
I was thinking that searching on a typed array for values will be faster in the worker, but I can try to use it like you suggested and see the impact